src/pybind/mgr/dashboard/frontend/src/app/shared/services/summary.service.spec.ts

```diff
@@ -66,14 +67,80 @@ describe('SummaryService', () => {
     subs.unsubscribe();
   }));
 
+  describe('Should test subscribe without initial value', () => {
+    beforeEach(() => {
+      summaryService['summaryDataSource'].next(undefined);
+    });
+
+    it('should call subscribeOnce', () => {
+      let result;
+      let i = 0;
+      const subscriber = summaryService.subscribeOnce((response) => {
+        i++;
+        result = response;
+      });
+
+      expect(subscriber).toEqual(jasmine.any(Subscriber));
+      expect(i).toBe(0);
+      expect(result).toEqual(undefined);
+
+      summaryService['summaryDataSource'].next(undefined);
+      expect(i).toBe(0);
+      expect(result).toEqual(undefined);
+      expect(subscriber.closed).toBe(false);
+
+      summaryService['summaryDataSource'].next(summary);
+      expect(result).toEqual(summary);
+      expect(i).toBe(1);
+      expect(subscriber.closed).toBe(true);
+
+      summaryService['summaryDataSource'].next(summary);
+      expect(result).toEqual(summary);
+      expect(i).toBe(1);
+    });
+
+    it('should call subscribe', () => {
+      let result;
+      let i = 0;
+      const subscriber = summaryService.subscribe((data) => {
+        i++;
+        result = data;
+      });
```

Comment:
You can set this in a before each in the describe block. Than you could also add a method to expect the state of these three variables, that would reduce the repetitiveness of the test code.

Reply:
Done.